### ISSUE TITLE
verification by short messages or email

### ISSUE THREAD

**Issue by sugionodjojo** [InstaPy/InstaPy#158]
when i try to login , always fail, the return message was wrong login data

I tried to manual login on application, and always got request to verified by email or sms

is it possible to add if asking to verify by sms or email always choose email, and return command to input verification code?

this could be the learning for me , and if there reading i can refer, i want to get into that.....

**Comment by sugionodjojo** [comment]
@vlntdds 

maybe something like this ?

```python
  #Check if user is logged-in (If there's two 'nav' elements)
  nav = browser.find_elements_by_xpath('//nav')
  if len(nav) == 2:
    return True
  else:
    action = ActionChains(browser).move_to_element(ReviewLoginForm_0)
```

**Comment by sugionodjojo** [comment]
still got wrong login data

i add this starting line 25 , login_util.py

def ReviewLoginForm_0():

    #Check if user is logged-in (If there's two 'nav' elements)
  nav = browser.find_elements_by_xpath('//nav')
  if len(nav) == 2:
    return True
  else:
    action = ActionChains(browser).move_to_element(ReviewLoginForm_0).click().perform()

I do not think the script click either verified by sms or by email button

is it okay to carry on on this page?

**Comment by sugionodjojo** [comment]
i think i have to leave this alone 
#Check if user is logged-in (If there's two 'nav' elements)

nav = browser.find_elements_by_xpath('//nav')
if len(nav) == 2:
return True
else:
return False

and instead change on instapy.py , start from line 74?

**Comment by eduardocalazansjr** [comment]
If there's user intervention needed, we can do nothing about.
If instagram flagged your account as a bot, we can do nothing about.

A little copy of the 2FA authentication, we can't just "wait for an user input" and check if the page was different in an amount of time, since the focus of the application was the use WITHOUT the chrome window.

When an exception like this raise in a console scenario, the user will can't do any intervention on the page and will process will get stuck (if he's using cronjob, he will have a bunch of python and chrome process stuck in the machine)

**Comment by sugionodjojo** [comment]
hello,

yes, without console, quick question,
so python cannot interact like Clang, when script can wait for input? ( set aside the headless chrome, as clang also run on terminal)

**Comment by eduardocalazansjr** [comment]
1. Yeah, it can interact, but in a large percent of cases it will get stuck for non-console users that use cronjob (or other automation tool) and will kill the "Automation Script Spirit", already discussed in the 2FA threads. 

2. The currently GUI (#44) is not for login, but for setup and creation of an initial script.

**Comment by sugionodjojo** [comment]
2FA  2 Factor authentication , 

thanks

**Comment by maverjk** [comment]
Today after troubling with settings and installation on a Centos server I finally got the first working results.
Unfortunately I discovered that is not supported 2 Factor Authentication.

Well I think that this features is quite important to keep and should be supported in some way. For what I know the security code should be asked just the first time then Instagram should store the device information to not ask it again with a new authentication request. I'm not pretty sure if that would be the same with a python script running on a server, however would be really nice to have this feature in the script asking to the user the code on the first authentication attempt.

Thanks for your work, it's really well done ;)
